linera-core/src/chain_worker/state.rs

```diff
@@ -1291,10 +1291,10 @@ where
                     .await
                     .with_execution_context(ChainExecutionContext::Query)?;
                 self.execution_state_cache
-                    .insert_owned(&requested_state, state);
+                    .insert_owned(&requested_block, state);
```

Comment:
This is still a state, though, and not a block, is it? It's just _keyed_ by block hash?

Reply:
Why? the key is still the block (hash).